py/torch_tensorrt/dynamo/lowering/_remove_sym_nodes.py

```diff
@@ -0,0 +1,30 @@
+import logging
+
+import torch
+
+logger = logging.getLogger(__name__)
+
+
+def remove_sym_nodes(gm: torch.fx.GraphModule) -> torch.fx.GraphModule:
+    """Remove sym_int placeholders which get inserted due to torch.compile's
+    dynamic=True behavior
+    """
+    # Extract SymInt placeholder Tensors
+    placeholders = [
+        node
+        for node in gm.graph.nodes
+        if (
+            node.op == "placeholder"
+            and isinstance(node.type, type)
+            and issubclass(node.type, torch.SymInt)
```

Comment:
Just in case this were to change in the future, could this be updated to include other symbolic types, such as `issubclass(node.type, (torch.SymInt, torch.SymFloat, torch.SymBool))`

Reply:
`torch.SymInt` nodes here indicate the shape of that dimension and don't seem to be used by other nodes (in torch.compile's dynamic=True behavior). I'm not sure when `torch.SymFloat` and `torch.SymBool` nodes show up, so we remove only `torch.SymInt` nodes conservatively. We could change this in the future based on the behavior of `torch.SymFloat` and `torch.SymBool` nodes.